src/services/diners.js

```diff
@@ -0,0 +1,71 @@
+import { axiosBaseQuery } from '../apis/gateway';
+import { getTagLists } from '../helpers/service-tag-list.helper';
+import { createApi } from '@reduxjs/toolkit/query/react';
+
+// Define a service using a base URL and expected endpoints
+const resourceName = 'diners';
+const dinerTags = getTagLists(resourceName);
+
+export const dinersApi = createApi({
+  reducerPath: `${resourceName}Api`,
```

Comment:
Can you establish a reducer for only a path?

Reply:
React Tool Kit Query use cache for the requests, what this createApi function do is define a reducer so all http calls get stored inside it, it will expire and get removed by default at 30 seconds, this is configurable.

So next time the service method is called if there is a record for that call inside the defined reducer, it will return that value instead of making the HTTP call. 

Look at this video for further references, its a 3 part video
https://www.youtube.com/watch?v=an2hyrNwCNE